Shopify/shopify-cli
# Drop the pre and build information before doing the comparison

Addresses [this discussion](https://github.com/Shopify/shopify-cli/pull/1945#discussion_r800156644)

### WHY are these changes introduced?
This PR softens the Ruby version checks to support any version in the range 3.0.x. Before this change, versions above 3.0.2 would have thrown a warning but now, they can use any version up to 3.1.0 (not included).

### WHAT is this pull request doing?
- Adding a new method to `Semantic::Version`, `dropping_pre_and_build` that returns the version dropping the `pre` and `build` information.

### How to test your changes?
- Check out the branch.
- Run `bin/console`.
- `Semantic::Version.new('3.1.0-preview').dropping_pre_and_build < Semantic::Version.new('3.1.0').dropping_pre_and_build` should yield `false`.

### Update checklist
